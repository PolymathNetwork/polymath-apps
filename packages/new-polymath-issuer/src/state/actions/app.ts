import { createStandardAction } from 'typesafe-actions';

export const setActiveTransactionGroup = createStandardAction(
  'APP/SET_ACTIVE_TRANSACTION_GROUP'
)<string>();

export const initializePolyClientStart = createStandardAction(
  'APP/INITIALIZE_POLY_CLIENT_START'
)();

export const initializePolyClientSuccess = createStandardAction(
  'APP/INITIALIZE_POLY_CLIENT_SUCCESS'
)();

export const initializePolyClientFailure = createStandardAction(
  'APP/INITIALIZE_POLY_CLIENT_FAILURE'
)<Error>();
