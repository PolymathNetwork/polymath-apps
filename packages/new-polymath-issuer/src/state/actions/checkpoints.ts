import { types } from '@polymathnetwork/new-shared';
import { createEntityActions } from '~/state/helpers/createEntityActions';

const { createAction, updateAction, deleteAction } = createEntityActions<
  types.CheckpointEntity
>('CHECKPOINTS');

export { createAction, updateAction, deleteAction };
