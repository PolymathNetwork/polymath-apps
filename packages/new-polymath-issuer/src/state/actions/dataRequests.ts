import { createStandardAction } from 'typesafe-actions';
import { RequestKeys, Fetcher } from '~/types';
import { types } from '@polymathnetwork/new-shared';

const invalidateRequest = createStandardAction('DATA_REQUESTS/INVALIDATE')<{
  requestKey: RequestKeys;
  args?: types.Pojo;
}>();

const cacheData = createStandardAction('DATA_REQUESTS/CACHE')<{
  requestKey: RequestKeys;
  args: types.Pojo;
  fetchedIds: string[];
}>();

/**
 * This action signals the saga in charge of fetching data from the blockchain
 * that a component needs certain data
 */
const requestData = createStandardAction('DATA_REQUESTS/REQUEST')<{
  fetcher: Fetcher;
}>();

const fetchDataStart = createStandardAction('DATA_REQUESTS/FETCH_START')<{
  requestKey: RequestKeys;
  args: types.Pojo;
}>();

const fetchDataFail = createStandardAction('DATA_REQUESTS/FETCH_FAIL')<{
  errorMessage: string;
  requestKey: RequestKeys;
  args: types.Pojo;
}>();

export {
  invalidateRequest,
  cacheData,
  requestData,
  fetchDataStart,
  fetchDataFail,
};
