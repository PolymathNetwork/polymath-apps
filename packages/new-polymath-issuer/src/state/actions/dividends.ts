import { types } from '@polymathnetwork/new-shared';
import { createEntityActions } from '~/state/helpers/createEntityActions';

const { createAction, updateAction, deleteAction } = createEntityActions<
  types.DividendEntity
>('DIVIDENDS');

export { createAction, updateAction, deleteAction };
