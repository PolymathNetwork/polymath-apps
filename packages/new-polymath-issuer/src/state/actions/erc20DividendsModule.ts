import { Erc20DividendsModuleEntity } from '~/types';
import { createEntityActions } from '~/state/helpers/createEntityActions';
import { createStandardAction } from 'typesafe-actions';

const { createAction, updateAction, deleteAction } = createEntityActions<
  Erc20DividendsModuleEntity
>('ERC20_DIVIDENDS_MODULES');

const enableModule = createStandardAction('ERC20_DIVIDENDS_MODULES/ENABLE')<{
  securityTokenSymbol: string;
}>();

export { createAction, updateAction, deleteAction, enableModule };
