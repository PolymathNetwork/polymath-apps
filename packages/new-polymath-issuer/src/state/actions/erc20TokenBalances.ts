import { types } from '@polymathnetwork/new-shared';
import { createEntityActions } from '~/state/helpers/createEntityActions';

const { createAction, updateAction, deleteAction } = createEntityActions<
  types.Erc20TokenBalanceEntity
>('ERC20_TOKEN_BALANCES');

export { createAction, updateAction, deleteAction };
