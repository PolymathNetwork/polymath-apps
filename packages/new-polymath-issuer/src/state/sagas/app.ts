import { call, put } from 'redux-saga/effects';
import {
  initializePolyClientStart,
  initializePolyClientFailure,
  initializePolyClientSuccess,
} from '~/state/actions/app';
import { polyClient } from '~/lib/polymath';
import { requireWallet } from './accessControl';

export function* initializePolyClient() {
  yield put(initializePolyClientStart());
  yield requireWallet();
  try {
    yield call(polyClient.connect);
    yield put(initializePolyClientSuccess());
  } catch (error) {
    yield put(initializePolyClientFailure(error.message));
    throw error;
  }
}
