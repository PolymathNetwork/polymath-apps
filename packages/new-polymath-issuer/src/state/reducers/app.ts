import { Reducer } from 'redux';
import { getType } from 'typesafe-actions';
import * as actions from '~/state/actions/app';
import { AppActions } from '~/state/actions/types';

export interface AppState {
  activeTransactionGroup?: string;
}
const initialState: AppState = {};

export const reducer: Reducer<AppState, AppActions> = (
  state = initialState,
  action
) => {
  switch (action.type) {
    case getType(actions.setActiveTransactionGroup): {
      return {
        ...state,
        activeTransactionGroup: action.payload,
      };
    }
    default: {
      return state;
    }
  }
};
