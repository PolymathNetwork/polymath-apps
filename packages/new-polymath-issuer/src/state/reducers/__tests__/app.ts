import { reducer } from '../app';
import * as actions from '~/state/actions/app';

describe('Reducer: app', () => {
  test('initialState', () => {
    expect(reducer(undefined, {} as any)).toMatchSnapshot();
  });

  test('setActiveSequence sets activeSequence', () => {
    const id = 'someId';
    const result = reducer(undefined, actions.setActiveSequence(id));
    expect(result.activeSequence).toEqual(id);
  });

  test('initializePolyClient actions correctly update state', () => {
    const startResult = reducer(undefined, actions.initializePolyClientStart());
    const successResult = reducer(
      startResult,
      actions.initializePolyClientSuccess()
    );
    const failResult = reducer(
      startResult,
      actions.initializePolyClientFailure(new Error('Some error'))
    );
    // Not yet finished initializing
    expect(startResult.polyClientInitialized).toEqual(false);
    expect(successResult.polyClientInitialized).toEqual(true);
    expect(failResult.polyClientInitialized).toEqual(false);
  });

  test('setChangingRoute sets changingRoute correctly ', () => {
    let result = reducer(undefined, actions.setChangingRoute(true));
    expect(result.changingRoute).toEqual(true);
    result = reducer(undefined, actions.setChangingRoute(false));
    expect(result.changingRoute).toEqual(false);
  });
});
