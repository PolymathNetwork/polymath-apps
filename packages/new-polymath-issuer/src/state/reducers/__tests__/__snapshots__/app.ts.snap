// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reducer: app initialState 1`] = `
Object {
  "changingRoute": false,
  "polyClientInitialized": false,
}
`;
